scribe-org/Scribe-Android
# feat: Add Jacoco coverage and report generation

### Contributor checklist



-   [N/A] I have tested my code with the `./gradlew lintKotlin detekt test` command as directed in the [testing section of the contributing guide](https://github.com/scribe-org/Scribe-Data/blob/main/CONTRIBUTING.md#testing)
- [x] I have tested my code (see test notes below)

---

### Description



This change adds the Jacoco plugin to the app/gradle.build.kts file with configuration to generate coverage reports.

Additionally, this adds a step to the unit test Github workflow to use the generated coverage XML to create a comment on PRs. This is not configured to fail the PR build. Details: https://github.com/marketplace/actions/jacoco-report

One thing I wasn't able to configure is coverage verification checks and build time.

I followed details provided in this Medium article on adding coverage to Android projects:  https://medium.com/@nachare.reena8/measure-your-codes-reach-integrating-jacoco-code-coverage-in-android-apps-with-kotlin-dsl-382d577864e7

#### Testing

`./gradlew tasks` output:

```
Reporting tasks
---------------
jacocoCoreDebugCodeCoverage - Generate Jacoco coverage reports for the CoreDebug build
jacocoCoreReleaseCodeCoverage - Generate Jacoco coverage reports for the CoreRelease build
jacocoFdroidDebugCodeCoverage - Generate Jacoco coverage reports for the FdroidDebug build
jacocoFdroidReleaseCodeCoverage - Generate Jacoco coverage reports for the FdroidRelease build
```

Verified `./gradlew jacocoCoreDebugCodeCoverage` generates files with expected content:
* build/reports/jacoco/jacocoCoreDebugCodeCoverage/jacocoCoreDebugCodeCoverage.xml
* build/reports/jacoco/jacocoCoreDebugCodeCoverage/html/index.html

The github action/workflow fails to post a comment to the PR when run. I'm not sure what's causing that. I've enabled debug logging on the action/workflow but someone with repo management permissions may need to help me understand what's going wrong with the token permissions.

### Related issue





-   #196
